app/smsbot/models.py

```
@@ -98,3 +108,36 @@ def send_sms(self, text, ignore_opt_out=False):
                         f"Failed to send via twilio (attempt {tries}): {e} ({msg})"
                     )
                     time.sleep(tries)
+
+
+class Blast(TimestampModel, UUIDModel):
+    description = models.TextField(null=True)
+    content = models.TextField(null=True)
+    sql = models.TextField(null=True)
+
+    class Meta(object):
+        ordering = ["-created_at"]
+
+    def __str__(self):
+        return f"Blast - {self.description} ({self.uuid})"
+
+    def enqueue_sms_blast(self):
+        from .tasks import send_blast_sms
+
+        logger.info(f"Enqueueing {self}")
+        num = 0
+        for n in Number.objects.raw(self.sql):
+            send_blast_sms.delay(str(n.pk), self.pk)
+            num += 1
```

This could be slow. Do we have any way to recover if this code fails early that won't result in the user getting another MMS?

---

1- The final task refuses to send a dup message within the same blast
2- I think we can fiddle with the sql statement to exclude records that have already been sent